crates/example-types/src/storage_types.rs

```diff
@@ -52,6 +56,17 @@ impl<TYPES: NodeType> Default for TestStorage<TYPES> {
     }
 }
 
+impl<TYPES: NodeType> TestStorage<TYPES> {
+    pub async fn proposals_cloned(
+        &self,
+    ) -> BTreeMap<TYPES::Time, Proposal<TYPES, QuorumProposal<TYPES>>> {
+        self.inner.read().await.proposals.clone()
+    }
+    pub async fn high_qc_cloned(&self) -> Option<QuorumCertificate<TYPES>> {
+        self.inner.read().await.high_qc.clone()
+    }
```

Comment:
Not necessarily against the naming here, but would the `move`'d return value here not imply that the data was cloned?

Reply:
I guess kind of, I took the naming convention from something I saw in `async_broadcast` but that's pretty flimsy.  I like it here because it tells the caller for sure a real clone is happening.  With this interface I guess it's impossible for much else to happen considering the types are not `Copy` but when you just read `storage.proposals().get()` you might think it's returning a ref